packages/unified-selection/src/unified-selection/SelectionStorage.ts

```diff
@@ -0,0 +1,237 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+/** @packageDocumentation
+ * @module UnifiedSelection
+ */
+
+import { Selectable, Selectables } from "./Selectable";
+
+/**
+ * An interface for selection change listeners.
+ * @beta
+ */
+export declare type StorageSelectionChangesListener = (args: StorageSelectionChangeEventArgs, storage: SelectionStorageInterface) => void;
+
+/**
+ * The type of selection change
+ * @beta
+ */
+export enum StorageSelectionChangeType {
+  /** Added to selection. */
+  Add,
+  /** Removed from selection. */
+  Remove,
+  /** Selection was replaced. */
+  Replace,
+  /** Selection was cleared. */
+  Clear,
+}
+
+/**
+ * The event object that is sent when the selection changes.
+ * @beta
+ */
+export interface StorageSelectionChangeEventArgs {
+  /** The name of the selection source which caused the selection change. */
+  source: string;
+  /** Level of the selection. */
+  level: number;
+  /** The selection change type. */
+  changeType: StorageSelectionChangeType;
+  /** Selectables affected by this selection change event. */
+  selectables: Selectables;
+  /** iModel key with which the selection is associated with. */
+  iModelKey: string;
+  /** The timestamp of when the selection change happened */
+  timestamp: Date;
+}
+
+/**
+ * Selection storage interface which provides main selection and sub-selection.
+ * @beta
+ */
+export interface SelectionStorageInterface {
+  /** A callback that is called when selection changes */
+  onSelectionChange: (event: StorageSelectionChangeEventArgs, storage: SelectionStorageInterface) => void;
```

Comment:
This won't support multiple event listeners. Instead, please define an `Event` interface and use that. See here: https://github.com/iTwin/presentation/blob/0ab2c5c4a51620c6bd2af3e39ef55cbaca9dcb79/packages/core-interop/src/core-interop/Transactions.ts#L6C1-L15C2.